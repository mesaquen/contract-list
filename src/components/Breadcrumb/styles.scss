.breadcrumb--container {
  display: flex;
}

.breadcrumb--item {
  display: flex;
  align-items: center;
}

.breadcrumb--separator {
  margin: 0 0.5rem;
  color: var(--secondary-yellow);
}
