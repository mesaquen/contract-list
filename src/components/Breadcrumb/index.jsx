import React from 'react'
import PropTypes from 'prop-types'
import Text from '../Typography'
import { MdChevronRight as Icon } from 'react-icons/md'
import './styles.scss'

const Breadcrumb = ({ data }) => {
  const renderItem = (item, index, items) => {
    const bold = index === items.length - 1
    const shouldRenderSeparator = index > 0
    const separator = shouldRenderSeparator ? (
      <Icon className="breadcrumb--separator" size="1.25rem" />
    ) : null

    return (
      <div className="breadcrumb--item" key={item}>
        {separator}
        <Text size={Text.S} bold={bold} variant="secondary">
          {item}
        </Text>
      </div>
    )
  }
  return <div className="breadcrumb--container">{data.map(renderItem)}</div>
}

Breadcrumb.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string),
}

export default Breadcrumb
