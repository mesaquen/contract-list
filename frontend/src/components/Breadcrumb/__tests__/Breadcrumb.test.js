import { cleanup, render } from '@testing-library/react'
import Breadcrumb from '../index'

beforeEach(cleanup)

describe('Breadcrumbs', () => {
  const history = ['Page1', 'Page2']

  test('matches snapshot', async () => {
    const { asFragment } = render(<Breadcrumb data={history} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
