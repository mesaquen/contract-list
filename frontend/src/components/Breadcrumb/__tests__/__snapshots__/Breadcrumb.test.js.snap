// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb--container"
  >
    <div
      class="breadcrumb--item"
    >
      <span
        class="typo--s typo--text-secondary"
      >
        Page1
      </span>
    </div>
    <div
      class="breadcrumb--item"
    >
      <svg
        class="breadcrumb--separator"
        fill="currentColor"
        height="1.25rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
      <span
        class="typo--s typo--bold typo--text-secondary"
      >
        Page2
      </span>
    </div>
  </div>
</DocumentFragment>
`;
